numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -1724,6 +1741,10 @@ mapiter_@name@(PyArrayMapIterObject *mit)
                         NPY_AUXDATA_FREE(transferdata);
                         return -1;
                     }
+                    if(subspace_ptrs[0] == self_ptr &&
+                       subspace_ptrs[1] == mit->extra_op_ptrs[0]) {
+                        skip <<= 1;
```

Comment:
Not sure why two iterations without checking closer, but can't the bit shift overflow to 0?